consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/payload/CommitPayload.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.qbft.payload;
+
+import org.hyperledger.besu.consensus.common.bft.ConsensusRoundIdentifier;
+import org.hyperledger.besu.consensus.common.bft.payload.Payload;
+import org.hyperledger.besu.consensus.qbft.messagedata.QbftV1;
+import org.hyperledger.besu.crypto.SECP256K1.Signature;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.rlp.RLPInput;
+import org.hyperledger.besu.ethereum.rlp.RLPOutput;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+
+public class CommitPayload implements Payload {
+  private static final int TYPE = QbftV1.COMMIT;
+  private final ConsensusRoundIdentifier roundIdentifier;
+  private final Hash digest;
+  private final Signature commitSeal;
+
+  public CommitPayload(
+      final ConsensusRoundIdentifier roundIdentifier,
+      final Hash digest,
+      final Signature commitSeal) {
+    this.roundIdentifier = roundIdentifier;
+    this.digest = digest;
+    this.commitSeal = commitSeal;
+  }
+
+  public static CommitPayload readFrom(final RLPInput rlpInput) {
+    rlpInput.enterList();
+    final ConsensusRoundIdentifier roundIdentifier = ConsensusRoundIdentifier.readFrom(rlpInput);
+    final Hash digest = Payload.readDigest(rlpInput);
+    final Signature commitSeal = rlpInput.readBytes(Signature::decode);
```

Comment:
does it make sense to wrap these in try/finally block (where finally block contains `rlpInput.leaveList()`)? Or this area is meant to be exception free always.

Reply:
nvm ... I misread the method ... its `readFrom` the rlpInput.